From e3d75f25ea2b920ef0e762796c97722b27735c94 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Thu, 13 Sep 2018 20:56:44 +0000
Subject: [PATCH] turn TLS off

---
 Makefile.rule | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.rule b/Makefile.rule
index 6457532..45ea6e5 100644
--- a/Makefile.rule
+++ b/Makefile.rule
@@ -113,7 +113,7 @@ USE_SIMPLE_THREADED_LEVEL3 = 1
 # thread-local storage instead of a central memory buffer in memory.c
 # Note that if your system uses GLIBC, it needs to have at least glibc 2.21
 # for this to work.
-USE_TLS = 1
+# USE_TLS = 1
 
 # If you want to drive whole 64bit region by BLAS. Not all Fortran
 # compiler supports this. It's safe to keep comment it out if you
-- 
2.18.0

